service-recruiter/src/main/java/com/iwa/recruiterservice/address/AddressController.java

```diff
@@ -0,0 +1,65 @@
+package com.iwa.recruiterservice.address;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/addresses")
+public class AddressController {
+    private final AddressService service;
+
+    public AddressController(AddressService service) {
+        this.service = service;
+    }
+
+    @GetMapping
+    public List<Address> getAllAddresses() {
+        return service.getAddresses();
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<Address> getAddressById(@PathVariable Long id) {
+        Address res = service.getAddressById(id);
+        if (res != null) {
+            return new ResponseEntity<>(res, HttpStatus.OK);
+        } else {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+    }
```

Comment:
you could improve code style:

return service.getAddressById(id)
                .map(res -> ResponseEntity.status(HttpStatus.OK).body(res))
                .orElse(new ResponseEntity<>(HttpStatus.NOT_FOUND));